core/src/test/java/org/elasticsearch/common/xcontent/ConstructingObjectParserTests.java

```diff
@@ -174,10 +175,10 @@ public void testRepeatedConstructorParam() throws IOException {
                 + "  \"vegetable\": 2\n"
                 + "}");
         Throwable e = expectThrows(ParsingException.class, () -> randomFrom(HasCtorArguments.ALL_PARSERS).apply(parser, MATCHER));
-        assertEquals("[has_required_arguments] failed to parse field [vegetable]", e.getMessage());
+        assertEquals("[has_required_arguments] failed to parse object", e.getMessage());
         e = e.getCause();
-        assertThat(e, instanceOf(IllegalArgumentException.class));
-        assertEquals("Can't repeat param [vegetable]", e.getMessage());
+        assertThat(e, instanceOf(JsonParseException.class));
```

Comment:
> So this is one problem with doing this as a read-on-startup property.

I agree that testing is not as simple as it could be but my priority was that the production code is dead-simple (i.e. no fancy abilities to change the JSON factory at runtime after initial creation). Given the fact that it's likely that we will enable this check without an ability to opt-out at some point in the future, I felt this was the right trade-off.